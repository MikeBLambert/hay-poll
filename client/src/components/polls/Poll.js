import React from 'react';
import PropTypes from 'prop-types';

export default function Poll({ title, choices }) {
  const choiceList = choices.map((choice, i) => {
    return <li key={i}>{choice}</li>;
  });
  return (
    <div>
      <h1>{title}</h1>
      <ul>{choiceList}</ul>
    </div>
  );
}

Poll.propTypes = {
  title: PropTypes.string.isRequired,
  choices: PropTypes.array.isRequired
};
