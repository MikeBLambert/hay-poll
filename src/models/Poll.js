import mongoose from 'mongoose';

const pollSchema = new mongoose.Schema({
  issue: {
    type: String,
    required: true
  },
  options: [{
    choice: {
      type: String,
      required: true
    }
  }]
});

pollSchema.methods.results = function() {
  return this.model('Vote').aggregate([
    { $match: { pollId: this._id } },
    { $unwind: '$votes' },
    { $group:
        { _id:
            { choiceId: '$votes.choiceId', choiceName: '$votes.choiceName' },
            totalVotes: { $sum: 1 }
        }
    }
  ])
}

export default mongoose.model('Poll', pollSchema)
