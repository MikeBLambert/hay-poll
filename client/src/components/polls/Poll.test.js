import React from 'react';
import Poll from './Poll';
import { shallow } from 'enzyme';

describe('Poll component', () => {
  it('renders a Poll component', () => {

    const choices = ['GOT', 'Seinfeld', 'The Simpsons'];
    const wrapper = shallow(<Poll title='Best Show Ever' choices={choices} />);
    expect(wrapper).toMatchSnapshot();
  });
});
