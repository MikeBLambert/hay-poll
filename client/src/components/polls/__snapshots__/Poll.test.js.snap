// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poll component renders a Poll component 1`] = `
<div>
  <h1>
    Best Show Ever
  </h1>
  <ul>
    <li
      key="0"
    >
      GOT
    </li>
    <li
      key="1"
    >
      Seinfeld
    </li>
    <li
      key="2"
    >
      The Simpsons
    </li>
  </ul>
</div>
`;
